objects/src/accounts/data.rs

```diff
@@ -0,0 +1,212 @@
+use std::{
+    fs::{self, File},
+    io::{self, Read},
+    path::Path,
+};
+
+use miden_crypto::utils::SliceReader;
+
+use super::{
+    super::utils::serde::{
+        ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable,
+    },
+    Account, Word,
+};
+
+// ACCOUNT DATA
+// ================================================================================================
+
+/// AccountData is a representation of the Account struct meant to be used
+/// for Account serialisation and deserialisation for transport of Account data
+#[derive(Debug, PartialEq, Eq)]
+pub struct AccountData {
+    pub account: Account,
+    pub account_seed: Option<Word>,
+    pub auth: AuthData,
+}
+
+impl AccountData {
+    pub fn new(account: Account, account_seed: Option<Word>, auth: AuthData) -> Self {
+        Self { account, account_seed, auth }
+    }
+
+    /// Serialises and writes binary AccountData to specified file
+    pub fn write(&self, filepath: &Path) -> Result<(), io::Error> {
+        fs::write(filepath, self.to_bytes())?;
+
+        Ok(())
+    }
+
+    /// Reads from file and tries to deserialise an AccountData
+    pub fn read(filepath: &Path) -> Result<Self, io::Error> {
+        let mut file = File::open(filepath)?;
+        let mut buffer = Vec::new();
+
+        file.read_to_end(&mut buffer)?;
+        let mut reader = SliceReader::new(&buffer);
+
+        Ok(AccountData::read_from(&mut reader).map_err(|_| io::ErrorKind::InvalidData)?)
+    }
+}
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for AccountData {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        let AccountData { account, account_seed, auth } = self;
+
+        let auth_scheme = match auth {
+            AuthData::RpoFalcon512Seed(_) => "RpoFalcon512",
+        };
+
+        let AuthData::RpoFalcon512Seed(auth_seed) = auth;
+
+        account.write_into(target);
+        match account_seed {
+            None => target.write_u8(0),
+            Some(seed) => {
+                target.write_u8(1);
+                seed.write_into(target)
+            },
+        };
+        let auth_scheme_len = auth_scheme.as_bytes().len();
+        target.write_u8(auth_scheme_len as u8);
+        auth_scheme.as_bytes().write_into(target);
+        auth_seed.write_into(target);
+    }
+}
+
+impl Deserializable for AccountData {
+    fn read_from<R: ByteReader>(
+        source: &mut R,
+    ) -> std::prelude::v1::Result<Self, DeserializationError> {
```

Comment:
I suppose this was generated by your IDE? It should be cleanup to just `Result<Self, DeserializationError>`